fe/fe-core/src/main/java/com/starrocks/clone/TabletSchedCtx.java

```diff
@@ -550,6 +551,9 @@ public void chooseSrcReplica(Map<Long, PathSlot> backendsWorkingSlots) throws Sc
             throw new SchedException(Status.UNRECOVERABLE, "unable to find source replica");
         }
 
+        // Shuffle the candidate list first so that we won't always choose the same replica with
```

Comment:
Why not remove the replica from the candidates?

Reply:
because it's an independent schedule every time, what you said need another refactor to the code.